### ISSUE TITLE
[linux] microphone keeps being opened and closed in lobby

### ISSUE THREAD

**Issue by benaryorg** [Regalis11/Barotrauma#10706]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

The microphone worked fine while going through the tutorial, the bar also indicated that the audio picked up and worked just fine with my mic.
As soon as I opened a multiplayer lobby (host server) the entire game started lagging heavily (i.e. laggy mouse pointer and UI), while audio kept playing fine.
After further examination it seems that a majority of CPU usage was spent on my audio daemon which was clearly overloaded, although it wasn't clear why until I opened my audio mixer and saw the mic input appearing and disappearing multiple times per second.
I'm assuming the game somehow gets into a loop of opening and closing the mic (because it seems to close properly) via the pulseaudio interface.
Disabling the microphone in the settings did not help, despite being disabled it's being opened and closed like no tomorrow.

Closing the game while in this state will cause a crash most of the time.

### Reproduction steps

1. run game on Linux (pulseaudio unix domain socket interface of pipewire)
2. host server

### Bug prevalence

Happens every time I play

### Version

0.19.14.0

### Which operating system did you encounter this bug on?

Linux

### Relevant error messages and crash reports

```shell
Barotrauma Client crash report (generated on 12/13/2022 19:24:10)


Barotrauma seems to have crashed. Sorry for the inconvenience!


11D783E09CC127D1BA37BC9FF6FBF709


Game version 0.19.14.0 (ReleaseLinux, branch release, revision 0daf5b87bd)
Graphics mode: 2560x1440 (BorderlessWindowed)
VSync ON
Language: English
Selected content packages: Vanilla
Level seed: no level loaded
Loaded submarine: None
Selected screen: Barotrauma.NetLobbyScreen
SteamManager initialized
Client (Round hadn't been started)


System info:
    Operating system: Unix 6.0.11.0 64 bit
    GPU name:
    Display mode: {Width:2560 Height:1440 Format:Color AspectRatio:1.7777778}
    GPU status: Normal


Exception: One or more errors occurred. (Operation canceled) (System.AggregateException)
Target site: Void ThrowIfExceptional(Boolean)
Stack trace:
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at Barotrauma.Networking.ChildServerRelay.ReadIncomingMsgs() in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Networking\ChildServerRelay.cs:line 111
   at Barotrauma.Networking.ChildServerRelay.<UpdateRead>g__readBytes|40_0(Span`1 readTo) in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Networking\ChildServerRelay.cs:line 168
   at Barotrauma.Networking.ChildServerRelay.UpdateRead() in <DEV>\Barotrauma\BarotraumaShared\SharedSource\Networking\ChildServerRelay.cs:line 160


InnerException: Operation canceled
Target site: Void MoveNext()
Stack trace:
   at System.IO.Pipes.PipeStream.ReadAsyncCore(Memory`1 destination, CancellationToken cancellationToken)
Last debug messages:
[12/13/2022 19:24:10] WARNING: Tried to restore enabled packages multiple times/without performing a backup
[12/13/2022 19:24:10] Exiting...
[12/13/2022 19:19:09] Lobby created!
[12/13/2022 19:17:06] WARNING: Warning: tried to complete the tutorial segment "tutorial.roles.captain.objective.station" in tutorial "tutorial.roles" but it isn't active!
[12/13/2022 19:08:16] WARNING: Warning: tried to remove the tutorial segment "droneobj_undock" in tutorial "tutorial.roles" but it isn't active!
[12/13/2022 18:56:47] Generating an outpost for the end of the level... (Location: Sarpedon Linea Outpost, level type: LocationConnection)
[12/13/2022 18:56:45] Loading a pre-selected outpost for the start of the level...
[12/13/2022 18:56:45] Level identifier: coldcavernstutorial
[12/13/2022 18:55:22] Attempting to open ALC device "Headphones"
[12/13/2022 18:55:13] Logged in as Katze (SteamID STEAM_1:0:42589855)
```

**Comment by benaryorg** [comment]
Additionally if I hit Escape while in the Host Server screen the reopening thing stops, and if I then go and reselect the mic in the settings I still see that everything's working as expected and it properly picks up the mic. As soon as I leave the menu though, it instantly reverts to opening and closing.

**Comment by benaryorg** [comment]
Okay, turns out this only happens in the lobby. Ingame everything works splendid. WTF.

**Comment by benaryorg** [comment]
To sum this up: this is not critical, it would be nice if it could be fixed easily, however the game is still playable anyway.

**Comment by Regalis11** [comment]
Fixed in https://github.com/Regalis11/Barotrauma-development/commit/dcccf8ab90db373f114a948f494fe6e5fe4c48c7

**Comment by benaryorg** [comment]
Can confirm that the issue is fixed with the latest update ([build id 10145301](https://steamdb.info/patchnotes/10145301/)).

Thank you very much!
